bash/t/t.functions

```diff
@@ -1,7 +1,46 @@
 # vim: et sr sw=4 ts=4 smartindent syntax=sh:
+#
+# t/t.functions
+#
+
+# $FAILURES:
+# ... used to capture failed test names.
 FAILURES=""
-SRC="${0##\./}"
-SRC="${SRC##t/}"
+
+# $SRC:
+# ... relative path to actual lib file to test
+#     determined by set_SRC(), called when you source
+#     t/t.functions.
+SRC=
+
+# tests must be run from within the ./bash dir and the test script
+# name should be set using a relative path
+set_SRC() {
+    if ! command -v realpath >/dev/null 2>&1
+    then
+        echo "ERROR: you must install GNU tools like realpath" >&2
+        echo "ERROR: before using opsgang/libs." >&2
+        exit 1
+    fi
+
+    if [[ ! $(pwd) =~ /bash$ ]]; then
```

Comment:
This isn't currently necessary, as the t/*/*.functions test scripts all try to source this helpers script (t/t.functions) using a relative path. i.e. They won't even get to source the file containing this function unless they are in the _bash_ parent dir.

However I've added it so when I create t/packer/packer_run.functions or whatever, it won't matter if I forget to use the path relative to the bash dir - this _if_ condition will catch it.